src/System.Private.ServiceModel/tests/Scenarios/Contract/Message/MessageContractTests.cs

```diff
@@ -119,4 +119,60 @@ public static void MessageBody_Elements_CustomerElement_Value_Matches()
 
         Assert.True(errorBuilder.Length == 0, errorBuilder.ToString());
     }
-}
\ No newline at end of file
+
+    [Fact]
+    [OuterLoop]
+    public static void MessageHeader_MustUnderstand_True()
+    {
+        StringBuilder errorBuilder = new StringBuilder();
+
+        try
+        {
+            var headers = MessageContractHelpers.SetupMessageContractTestsAndGetHeaders("MessageContractRequestReplyWithMessageHeader");
+            int index = headers.FindHeader(MessageContractConstants.extraValuesName, MessageContractConstants.extraValuesNamespace);
+            var header = headers[index];
+            if (header == null)
+            {
+                errorBuilder.AppendLine("There's no header in the message.");
```

Comment:
NIT: wondering if we should just use `Assert.True(header == null, "...");` in all the newly written stuff